src/otg/datasource/finngen/finngen_finemapping.py

```diff
@@ -0,0 +1,234 @@
+# pylint: disable=unsubscriptable-object
+"""Datasource ingestion: FinnGen Finemapping results (SuSIE) to studyLocus object."""
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+import pyspark.sql.functions as f
+import pyspark.sql.types as t
+from pyspark.sql import SparkSession, Window
+
+from otg.common.utils import parse_pvalue
+from otg.dataset.study_locus import StudyLocus
+
+if TYPE_CHECKING:
+    pass
+
+logging.basicConfig(level=logging.INFO, format="%(asctime)s %(message)s")
+
+
+@dataclass
+class FinnGenFinemapping:
+    """SuSIE finemapping dataset for FinnGen."""
+
+    @classmethod
+    def from_finngen_susie_finemapping(
+        cls: type[FinnGenFinemapping],
+        spark: SparkSession,
+        finngen_finemapping_df: (str | list[str]),
+        finngen_finemapping_summaries: (str | list[str]),
+        finngen_release_prefix: str,
+    ) -> StudyLocus:
+        """Process the SuSIE finemapping output for FinnGen studies.
+
+        Args:
+            spark (SparkSession): Spark session object.
+            finngen_finemapping_df (str | list[str]): SuSIE finemapping output filename(s).
+            finngen_finemapping_summaries (str | list[str]): filename of SuSIE finemapping summaries.
+            finngen_release_prefix (str): FinnGen study prefix.
+
+        Returns:
+            StudyLocus: Processed SuSIE finemapping output in StudyLocus format.
+        """
+        processed_finngen_finemapping_df = (
```

Comment:
It's usually a better practice to avoid reading data in the functions that contain business logic, so it's easier to test and debug. You read data in the step, and then call your function injecting the dependencies. Ideally you shouldn't need `spark` here.

Reply:
+1